src/graphs/hld.rs

```diff
@@ -151,7 +152,7 @@ impl HLD {
     /// - Time: O(1)
     /// - Space: O(1)
     pub fn in_sub(&self, u: usize, v: usize) -> bool {
-        u == v || self.sub_tree(u).contains(&self.tin[v])
+        (self.tin[u]..self.tin[u] + self.siz[u]).contains(&self.tin[v])
```

Comment:
I basically wanted methods to not depend on `vals_edges`, e.g. have the same behaviour whether `vals_edges` is true or not.